### ISSUE TITLE
[Feature Request] Show decorations in collapsed regions

### ISSUE THREAD

**Issue by kycutler** [microsoft/monaco-editor#3016]
### Context

- [X] This issue is not a bug report. *(please use a different template for reporting a bug)*
- [X] This issue is not a duplicate of an existing issue. *(please use the [search](https://github.com/microsoft/monaco-editor/issues) to find existing issues)*

### Description

I want to ensure that some decorators are visible in the editor even if the code regions are collapsed.

It looks like [VS Code has this ability](https://github.com/microsoft/vscode/blob/d54c705f6567958a732ac88b1c3ec4d2303fb026/src/vs/editor/common/model.ts#L112), but it isn't surfaced by Monaco.

Would be a helpful option to have available.

**Comment by hediet** [comment]
Related: https://github.com/microsoft/monaco-editor/issues/3005

It seems like decorations are not hidden when folded?

**Comment by kycutler** [comment]
To ensure visibility / discoverability of the decoration at all times. Originally this was for implementing custom, multi-editor find & replace functionality, but it could also be useful for e.g. debugging implementations, so the user can always be aware of where the current line is (even in a collapsed region).

For the find & replace issue I figured out it was easiest to just select the result in order to ensure its visibility (any collapsed regions will be expanded), but zero-width ranges are still an issue since the `showIfCollapsed` property (linked above) is not exposed in the Monaco typings. I figured out I can just ignore the typings and pass it anyway, but it would be great if that flag was exposed.
